app/javascript/article-form/articleForm.jsx

```diff
@@ -343,6 +355,28 @@ export default class ArticleForm extends Component {
           helpPosition={helpPosition}
           version={version}
         />
+        {this.state.showModal && (
+          <Modal title="You have unsaved changes" onClose={this.toggleModal}>
+            <div>
+              <p>
+                You've made changes to your post. Do you want to navigate to
+                leave this page?
+              </p>
+              <ButtonGroup className="crayons-article-form__modal__button-group">
+                <Button variant="danger" onClick={this.navigateHome}>
+                  Yes, go back
```

Comment:
It's a bit misleading label I think since it suggests I'll stay on the page, no? :D (although, _I know nothing_ 🇵🇱)
How about "Yes, leave the page"?

Reply:
I agree - it suggests to me "go back to the post I was editing" 

I think copy-wise we should explicitly mention they'll lose their edits e.g. "Do you want to leave this page and abandon your unsaved changes?" -> "Yes, abandon" (or something along those lines).

Reply:
I think they are not going to lose changes (afaik) because we autosave the form anyway.. this prompt is more to prevent accidentally leaving the page, I believe.

Reply:
TIL! I see we cache it in the browser, so I guess as long as they use the same browser the edits aren't lost, even if they don't show up yet in their saved draft preview.

Feels like a strange flow to me now though - we warn a user that they have unsaved changes (not exactly true), but then even if they click to say "sure let's leave the page anyway", we still persist those changes. But yeah if this is just to prevent accidental navigation then your suggestion of "Yes, leave the page" makes sense. 

Personally for me though, as a user I would think this modal is telling me I'm about to lose edits and I'd be a bit confused (as I clearly was when I wrote that last comment 😅 )

Reply:
@aitchiss I completely agree! I really struggled with the copy here because it *does* save. So it's a weird experience. Do you want to leave? It's going to save anyway but we are warning you.
However, a user expressed it was confusing to click the `x` with no warning, which I agree with!

Reply:
i guess it's similar to hitting CMD+Q in Chrome - you have to hold it in case you accidentally pressed it 🤷‍♂️

Reply:
Correct about the caching. It gets stored in local storage as long as they're using the same browser.

https://github.com/forem/forem/blob/ec57e15210c1e1a3abb1e0b1d0ab3da79bb992dd/app/javascript/article-form/articleForm.jsx#L123-L136

https://github.com/forem/forem/blob/ec57e15210c1e1a3abb1e0b1d0ab3da79bb992dd/app/javascript/article-form/articleForm.jsx#L59-L64

Reply:
The wording could be `Continue editing` like we have in comments.

![A screenshot of a comment being edited on dev.to](https://user-images.githubusercontent.com/833231/105894326-27358a80-5fe2-11eb-947c-cb1b2231e82b.png)